src/core/rattle.cpp

```diff
@@ -49,16 +49,17 @@ int n_rigidbonds = 0;
 
 /** Calculates the corrections required for each of the particle coordinates
     according to the RATTLE algorithm. Invoked from \ref correct_pos_shake()*/
-void compute_pos_corr_vec(int *repeat_);
+void compute_pos_corr_vec(int *repeat_, const ParticleRange &particles);
```

Comment:
Same comment as for `force_calc` applies, also the range should be reused withing thre function.